wagtail/wagtailadmin/static_src/wagtailadmin/scss/core.scss

```diff
@@ -417,19 +417,76 @@ footer,
             display: none;
         }
 
-        img {
-            width: 60px;
+        .wagtail-logo-container {
             float: none;
             margin: auto;
             display: block;
+            width: 60px;
+            height: 75px;
+            position: relative;
+            transition: all 0.25s cubic-bezier(0.28, 0.15, 0, 2.1);
+
+            img {
+                left: 0;
+                top: 0;
+                width: 60px;
+                position: absolute;
+                transition: all 0.25s cubic-bezier(0.28, 0.15, 0, 2.1);
+            }
+
+            img.wagtail-logo {
+                display: none; // This is the static logo (mobile only); use the animated one on desktop
```

Comment:
I think all of the "hidden on mobile" behaviour needs to be moved to utility classes so it's less intertwined with the other animation code. This could be defined in https://github.com/wagtail/wagtail/blob/master/client/scss/_utilities.scss, reusing https://github.com/wagtail/wagtail/blob/master/client/scss/_tools.breakpoints.scss. Reusing the nomenclature from above, this would look like:

```html
<div class="wagtail-logo-container">
    <img class="wagtail-logo wagtail-logo__full u-hidden@medium" />
    <img class="wagtail-logo wagtail-logo__body u-hidden@small" />
    <img class="wagtail-logo wagtail-logo__tail u-hidden@small" />
    <img class="wagtail-logo wagtail-logo__eyeopen u-hidden@small" />
    <img class="wagtail-logo wagtail-logo__eyeclosed u-hidden@small" />
</div>
```

...with some combination of the following utility classes:

```scss
.u-hidden\@medium {
    @include medium {
        display: none;
    }
}

.u-hidden\@small {
    @include small {
        display: none;
    }
}

.u-inline\@medium {
    @include medium {
        display: inline;
    }
}

.u-inline\@small {
    @include small {
        display: inline;
    }
}
```